File: Problem009/sol.java
/**
 * Special Pythagorean triplet
 * Problem 9
 * 
 * A Pythagorean triplet is a set of three natural numbers, a < b < c, for
 * which, a2 + b2 = c2 For example, 32 + 42 = 9 + 16 = 25 = 52. There exists
 * exactly one Pythagorean triplet for which a + b + c = 1000. Find the product
 * abc.
 */
public class sol {
	public static void main(String[] args) {
		double c = 0.0;
		int result = 0;
		
		for (int a = 0; a < 1000; a++)
			for (int b = 0; b < 1000; b++) {
				c = Math.sqrt((a * a) + (b * b));
				
				if (a + b + c == 1000 && a * b * c != 0) {
					result = (int) (a * b * c);
					break;
				}
			}

		System.out.println(result);
	}
}

File: Problem003/sol.cs
using System.IO;
using System;

class sol {
	static void Main(string[] args) {
		double num = 600851475143;
        	int result = 0;

		for(int i = 3; i <= Math.Sqrt(num); i += 2) {
			if(num % i == 0 && isprime(i) == 1) {
                		result = i;
			}
		}

        	Console.WriteLine(result);
	}

	static int isprime(double p) {
		if (p % 2 == 0) {
			return 0;
		}

		for (int i = 3; i <= Math.Sqrt(p); i += 2) {
			if (p % i == 0) {
				return 0;
			}
		}

		return 1;
	}
}
